client/apis/volumesnapshot/v1/types.go

```diff
@@ -108,13 +111,15 @@ type VolumeSnapshotSource struct {
 	// created.
 	// This field is immutable.
 	// +optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Spec.Source.PersistentVolumeClaimName is immutable"
```

Comment:
I believe that doing this we would have a better error message:

```suggestion
	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="persistentVolumeClaimName is immutable"
```

The original was:

```
The VolumeSnapshot "new-snapshot-test" is invalid: spec.source.persistentVolumeClaimName: Invalid value: "string": Spec.Source.PersistentVolumeClaimName is immutable
```

Reply:
I was trying to match exactly with the previous message and I agree with only showing field name makes sense since we will have fieldpath returned. Updated for all places. Thanks